ASM/c/dungeon_info.c

```diff
@@ -310,23 +336,52 @@ void draw_dungeon_info(z64_disp_buf_t* db) {
             // Draw small key counts
             sprite_load(db, &quest_items_sprite, 17, 1);
 
+            // First draw precompleted dungeons keys grayed out.
```

Comment:
Something similar here too as I said for the names. You should avoid iterating over a whole array multiple times.